### PR TITLE
add key binding to copy code

### PR DESCRIPTION
**Pull request by Sneezry**
Press arrow keys to select a code, press `enter` to copy.

Animation

### CODE REVIEW

File: src/components/Popup/MainBody.vue [hunk 1]
```diff
@@ -90,6 +90,87 @@ export default Vue.extend({
...
     },
+    isEntryVisible(entry: OTPEntry, shouldFilter: boolean, filter: boolean) {
+      return (
+        this.isSearchedEntry(entry) &&
+        (entry.pinned || !shouldFilter || !filter || this.isMatchedEntry(entry))
+      );
+    },
+    getTabindex(
+      entry: OTPEntry,
+      entries: OTPEntry[],
```
Comment (mymindstorm) [inline src/components/Popup/MainBody.vue:101]: entries, shouldFilter, and filter are all computed properties and don't need to be passed as arguments.

File: src/components/Popup/MainBody.vue [hunk 2]
```diff
@@ -90,6 +90,87 @@ export default Vue.extend({
...
+    ) {
+      const firstEntry = entries.find((entry) =>
+        this.isEntryVisible(entry, shouldFilter, filter)
+      );
+
+      return entry === firstEntry ? 0 : -1;
+    },
+    findNextEntryIndex(
+      entries: OTPEntry[],
+      shouldFilter: boolean,
```
Comment (mymindstorm) [inline src/components/Popup/MainBody.vue:113]: same as above.

File: src/components/Popup/MainBody.vue [hunk 3]
```diff
@@ -90,6 +90,87 @@ export default Vue.extend({
...
+      filter: boolean
+    ) {
+      const firstEntry = entries.find((entry) =>
+        this.isEntryVisible(entry, shouldFilter, filter)
+      );
+
+      return entry === firstEntry ? 0 : -1;
+    },
+    findNextEntryIndex(
+      entries: OTPEntry[],
```
Comment (mymindstorm) [inline src/components/Popup/MainBody.vue]: On all functions, entries, shouldFilter, and filter are all computed properties and don't need to be passed as arguments.

**Comment by mymindstorm** [comment]
I really like this!

Bugs / Other:
- When entering password, it does not function. Need to hit tab or reopen the extension for it to work.
- If I click anywhere else (even an entry) it will not let me use arrow keys until tab or reopen. 
	- Should we listen for keys on the document?
	- Should we focus entries on click?

File: src/components/Popup/MainBody.vue (hunk 3 above)
Comment (Sneezry) [reply src/components/Popup/MainBody.vue]: Updated.

**Comment by Sneezry** [comment]
> When entering password, it does not function. Need to hit tab or reopen the extension for it to work.

Fixed. After applying password, the first entry will be auto focused.

> If I click anywhere else (even an entry) it will not let me use arrow keys until tab or reopen.

Entry will have focus when it is clicked. If you click anywhere else, the entry should not have focus. And is as designed.

> Should we listen for keys on the document?

I don't think so. By following w3 accessibility guide, `tab` should be used to select active element, and arrow keys should be used to select items in an active group element. If the group element is not active, we should not response to the action.

> Should we focus entries on click?

Yes. This is a bug in previous commit. I have fixed that in the recent commit.

File: src/components/Popup/MainBody.vue [hunk 4]
```diff
@@ -14,6 +14,7 @@
         v-model="searchText"
         v-bind:placeholder="i18n.search"
         type="text"
+        tabindex="-1"
```
Comment (mymindstorm) [inline src/components/Popup/MainBody.vue:17]: I think this should be able to be tabbed to, but it is ok to not allow it until we get around to overhaling keyboard nav for the header and menu. Hopefully compoentizing everything will help at the second version of that.

**Review by mymindstorm — APPROVED** [review]
Looks great

**Comment by mymindstorm** [comment]
Will this supersede #689?

**Comment by Sneezry** [comment]
> Will this supersede #689?

I suppose. Number keys could be hard to use to copy a specific entry. Imaging how to copy the 7th entry in the list. I have to count entries from top to bottom.

**Pull request merged**